runtime/commonMain/src/kotlinx/benchmark/BenchmarkDescriptor.kt

```diff
@@ -1,7 +1,21 @@
 package kotlinx.benchmark
 
-open class BenchmarkDescriptor<T>(
+abstract class BenchmarkDescriptor<T>(
     val name: String,
     val suite: SuiteDescriptor<T>,
-    val function: T.() -> Any?
-)
\ No newline at end of file
+    val blackhole: Blackhole,
+)
+
+open class BenchmarkDescriptorWithNoBlackholeParameter<T>(
```

Comment:
BTW, why blackhole property is needed in `NoBlackhole` variant?

Reply:
This is NoBlackhole_Parameter_ variant. The blackhole still needed for benchmark return value consumption.